File: app/src/main/java/huce/fit/appreadstories/dialog/BaseDialog.java
package huce.fit.appreadstories.dialog;

import android.app.Dialog;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.view.Window;
import android.view.WindowManager;

public class BaseDialog implements BaseDialogView {

    public Dialog mDialog;
    public Context mContext;

    protected BaseDialog(Context context) {
        mContext = context;
    }

    public void setDialog(int layoutId) {
        mDialog = new Dialog(mContext);
        mDialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        mDialog.setContentView(layoutId);
        // true click bên ngoài dialog có thể tắt dialog
//        mDialog.setCancelable(false);
    }

    @Override
    public void setWindow(int width, int height, int gravity, int animation) {
        Window window = mDialog.getWindow();
        if (window == null) {
            return;
        }

        //chiều dài, chiều cao dialog
        window.setLayout(width, height);
        //background
        window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));

        // vị trí dialog
        WindowManager.LayoutParams windowAttributes = window.getAttributes();
        windowAttributes.gravity = gravity;
        if (animation != 0) {
            windowAttributes.windowAnimations = animation;
        }
        window.setAttributes(windowAttributes);
    }

    @Override
    public void show() {
        mDialog.show();
    }

    @Override
    public void dismiss() {
        mDialog.dismiss();
    }
}

File: app/src/main/java/huce/fit/appreadstories/dialog/BaseDialog.kt
package huce.fit.appreadstories.dialog

import android.app.Dialog
import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.view.Window

open class BaseDialog(val context: Context) : BaseDialogView {

    val dialog = Dialog(context)

    fun setDialog(layoutId: Int) {
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        dialog.setContentView(layoutId)
        // true click bên ngoài dialog có thể tắt dialog
//        dialog.setCancelable(false);
    }

    final override fun setWindow(width: Int, height: Int, gravity: Int, animation: Int) {
        val window = dialog.window ?: return

        //chiều dài, chiều cao dialog
        window.setLayout(width, height)
        //background
        window.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))

        // vị trí dialog
        val windowAttributes = window.attributes
        windowAttributes.gravity = gravity
        if (animation != 0) {
            windowAttributes.windowAnimations = animation
        }
        window.attributes = windowAttributes
    }

    override fun show() = dialog.show()

    override fun dismiss() = dialog.dismiss()
}
